straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,133 @@
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.24'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U9'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = straxen.n_hard_aqmon_start
+        aqmon_channel_names = ('sum_wf','m_veto_sync',
+                               'hev_stop', 'hev_start', 'he_stop','he_start','busy_stop', 'busy_start')
+        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
+        
+        # Keeping ch_range as class attribute for now, maybe will need it for other features
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
+        self.veto_names = ['busy_','he_','hev_']
+    
+    def get_window_size(self):
+        # Give a very wide window
+        return (self.config['max_veto_gap']*100)
+    
+    def compute(self, aqmon_hits): 
+        hits = aqmon_hits
+        
+        # Make a dict to populate later
+        res_temp = dict()
+        
+        for i, veto in enumerate(self.veto_names):
+            veto_hits = channel_select(hits, self.channel_map[veto + 'stop'], self.channel_map[veto + 'start'])
+            vetos = self.merge_vetos(veto_hits, gap = self.config['max_veto_gap'],\
+                                           dtype = strax.time_fields, t = 0)
+            
+            # Check that we found a veto interval and update the resulting dict
+            if len(vetos):
+                res_temp.setdefault("time",[]).extend(vetos['time'])
+                res_temp.setdefault("endtime",[]).extend(vetos['endtime'])
+                res_temp.setdefault("veto_interval",[]).extend((vetos['endtime'] - vetos['time']))
+                res_temp.setdefault("veto_type",[]).extend([veto + 'veto']*len(vetos))
+        
+        res = strax.dict_to_rec(res_temp)
```

Comment:
There seems to be an error with empty dicts: 

```
x = {}, dtype = None

    @export

    def dict_to_rec(x, dtype=None):

        """Convert dictionary {field_name: array} to record array

        Optionally, provide dtype

        """

        if isinstance(x, np.ndarray):

            return x

    

        if dtype is None:

            if not len(x):

>               raise ValueError("Cannot infer dtype from empty dict")

E               ValueError: Cannot infer dtype from empty dict
```

Reply:
```python
res = strax.dict_to_rec(res_temp, dtype = self.dtype)
```

Reply:
Yes, sorry forgot to copy the line

Reply:
OK, yeah it's the trivial solution to fix Travis. Alex I just added it in d92de8c. I believe the review has been quite thorough so let's merge it! Would love to have this shown on the website.

Reply:
@WenzDaniel @jorana Cheers guys :)